src/Processors/Formats/Impl/NpyRowInputFormat.cpp

```diff
@@ -0,0 +1,481 @@
+#include <Common/Exception.h>
+#include <Columns/ColumnFixedString.h>
+#include <Core/TypeId.h>
+#include <Core/Types_fwd.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnArray.h>
+#include <Columns/ColumnsNumber.h>
+#include <Storages/IStorage.h>
+#include <Columns/IColumn.h>
+#include <Core/Field.h>
+#include <Core/NamesAndTypes.h>
+#include <DataTypes/DataTypeArray.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <DataTypes/IDataType.h>
+#include <DataTypes/Serializations/ISerialization.h>
+#include <IO/ReadBuffer.h>
+#include <IO/WriteHelpers.h>
+#include <Processors/Formats/IRowInputFormat.h>
+#include <base/types.h>
+#include <boost/algorithm/string/split.hpp>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int INCORRECT_DATA;
+    extern const int BAD_ARGUMENTS;
+    extern const int LOGICAL_ERROR;
+    extern const int UNKNOWN_TYPE;
+    extern const int ILLEGAL_COLUMN;
+}
+
+namespace
+{
+
+DataTypePtr getDataTypeFromNumpyType(const std::shared_ptr<NumpyDataType> & numpy_type)
+{
+    switch (numpy_type->getTypeIndex())
+    {
+        case NumpyDataTypeIndex::Int8:
+            return std::make_shared<DataTypeInt8>();
+        case NumpyDataTypeIndex::Int16:
+            return std::make_shared<DataTypeInt16>();
+        case NumpyDataTypeIndex::Int32:
+            return std::make_shared<DataTypeInt32>();
+        case NumpyDataTypeIndex::Int64:
+            return std::make_shared<DataTypeInt64>();
+        case NumpyDataTypeIndex::UInt8:
+            return std::make_shared<DataTypeUInt8>();
+        case NumpyDataTypeIndex::UInt16:
+            return std::make_shared<DataTypeUInt16>();
+        case NumpyDataTypeIndex::UInt32:
+            return std::make_shared<DataTypeUInt32>();
+        case NumpyDataTypeIndex::UInt64:
+            return std::make_shared<DataTypeUInt64>();
+        case NumpyDataTypeIndex::Float32:
+            return std::make_shared<DataTypeFloat32>();
+        case NumpyDataTypeIndex::Float64:
+            return std::make_shared<DataTypeFloat64>();
+        case NumpyDataTypeIndex::String:
+            return std::make_shared<DataTypeString>();
+        case NumpyDataTypeIndex::Unicode:
+            return std::make_shared<DataTypeString>();
+    }
+    throw Exception(ErrorCodes::UNKNOWN_TYPE, "Numpy type {} is not supported", magic_enum::enum_name(numpy_type->getTypeIndex()));
+}
+
+DataTypePtr createNestedArrayType(const DataTypePtr & nested_type, size_t depth)
+{
+    DataTypePtr result_type = nested_type;
+    assert(depth > 0);
+    if (depth > 1)
+    {
+        for (size_t i = 0; i < depth - 1; ++i)
+            result_type = std::make_shared<DataTypeArray>(std::move(result_type));
+    }
+    return result_type;
+}
+
+size_t parseTypeSize(const std::string type)
+{
+    try
+    {
+        size_t size = std::stoi(type);
+        return size;
+    }
+    catch (...)
+    {
+        throw Exception(ErrorCodes::INCORRECT_DATA, "Invalid data type");
+    }
+}
```

Comment:
```suggestion
size_t parseTypeSize(const std::string & size_str)
{
    try
    {
        size_t size = std::stoi(size_str);
        return size;
    }
    catch (...)
    {
        throw Exception(ErrorCodes::INCORRECT_DATA, "Invalid data type size: {}", size_str);
    }
}
```

Reply:
I see you still use `std::stoi`. Is there a problem sith our `tryReadIntText`?

Reply:
done